howto/android/odm_vhal.md

```diff
@@ -0,0 +1,128 @@
+*since 1.22.0*
+
+[note type="information" status="Note"]Replacing the Anbox VHAL is only supported on AAOS images.
+The Anbox Cloud dashboard does not support custom VHAL implementations.[/note]
+
+This document will guide through the process of replacing the Anbox Cloud VHAL
+implementation with your own implementation placed in the
+[ODM partition](https://source.android.com/docs/core/architecture/partitions/odm-partitions)
+using an [addon](https://anbox-cloud.io/docs/explanation/addons).
+
+Your custom VHAL implementation must provide an
+[HIDL VHAL interface](https://source.android.com/docs/automotive/vhal/hidl-vhal-interface)
+and be built against VNDK 29.
+
+## Create the addon
+
+This addon will be tasked with the following:
+
+1. Copying the VHAL binary to `/odm/bin/hw`
+1. Copying an
+[init file](https://android.googlesource.com/platform/system/core/+/master/init/README.md)
+for your VHAL service to `/odm/etc/init`
+1. Disabling the Anbox Cloud VHAL by setting the `ro.anbox.automotive.vhal`
+property to `odm`. This will be done by writing that system property to the
+`/odm/etc/build.prop` file.
+
+In the following, we will name our addon `custom-vhal`.
```

Comment:
Did you mean to say "In the following example"?

Reply:
Yeah, that seems better, thanks!